src/main/java/com/puppycrawl/tools/checkstyle/filters/XpathFilterElement.java

```diff
@@ -171,13 +144,29 @@ public XpathFilterElement(Pattern files, Pattern checks, Pattern message,
                 throw new IllegalArgumentException("Incorrect xpath query: " + xpathQuery, ex);
             }
         }
+        allPropertiesAreNull = fileRegexp == null
+                             && checkRegexp == null
+                             && messageRegexp == null
+                             && moduleId == null
+                             && xpathExpression == null;
     }
 
     @Override
     public boolean accept(TreeWalkerAuditEvent event) {
         return !isFileNameAndModuleAndModuleNameMatching(event)
                 || !isMessageNameMatching(event)
-                || !isXpathQueryMatching(event);
+                || !isXpathQueryMatching(event)
+                || areAllPropertiesNull();
+    }
+
+    /**
+     * Check if all properties are set to null.
+     *
+     * @return true if all properties are set to null
+     */
+
+    private boolean areAllPropertiesNull() {
```

Comment:
I see no reason for private getter. You can access field directly, without getter

Reply:
Done